### ISSUE TITLE
RecyclerView to Pager animation and data changes in RecyclerView

### ISSUE THREAD

**Issue by MFlisar** [alexvasilkov/GestureViews#61]
If you allow to add items to the `ViewPager` (and in parallel to the `RecyclerView`) while the `ViewPager` is visible, the `ViewsTransitionAnimator` does not update it's source view correctly. 

I think the tracker should add a `registerAdapterDataObserver` to the `RecyclerView` and make sure, that it reevalutes it's state if a a change occurs, which means updating the from view according to the current id...

What do you think about that?

**Comment by alexvasilkov** [comment]
It should be enough to listen for children updates using `RecyclerView.OnChildAttachStateChangeListener`.

Have you experienced a real issue where source view is not updated, or your concern is theoretical?

**Comment by MFlisar** [comment]
I'm doing following:

1) ViewPager:
* add an item to the ViewPager (add item to adapter so that it is replacing the current one at the current position) and notifying everything about the change
2) RecyclerView:
* add an item to it and call `notifyItemInserted`

The logs show, that afteradding an item to the `ViewPager` the `ViewsTracker` is returning the wrong id in `getIdForPosition` because the adapter is not updated yet.

I'm seeing similar issues when deleting items (the visibility of the from view is not updated)... So I assume, I must tell the animator to update it's state when I add/delete items from the `RecyclerView` to make sure it's up to date

**Comment by MFlisar** [comment]
Following helps:

After I know that the `RecyclerView` and the `ViewPager` are updated, I call `animator.enter(id, false)`.This solves most of the problems, only the swipe close is not always working... Whereas the back animation always works...

**Comment by alexvasilkov** [comment]
I reproduced the issue where exit animation is not working once current page is removed, and fixed it (but didn't push yet).

But I tested deletion from both recycler view and view pager (deleting current page) and it works fine. So it seems like something wrong with your ViewsTracker.

I also made few minor fixes recently in transition listeners, so maybe your issue was fixed their, but I don't think they are related.
